// @keystonejs/fields/types/Select/views/Controller/dist/fields.cjs.prod.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: !0
});

var _defineProperty = require("@babel/runtime/helpers/defineProperty"), _objectWithoutProperties = require("@babel/runtime/helpers/objectWithoutProperties");

require("@babel/runtime/helpers/classCallCheck"), require("lodash.isequal");

var Controller_dist_fields = require("../../../../../Controller/dist/fields.cjs.prod.js");

function _interopDefault(e) {
  return e && e.__esModule ? e : {
    default: e
  };
}

var _defineProperty__default = _interopDefault(_defineProperty), _objectWithoutProperties__default = _interopDefault(_objectWithoutProperties);

function ownKeys(object, enumerableOnly) {
  var keys = Object.keys(object);
  if (Object.getOwnPropertySymbols) {
    var symbols = Object.getOwnPropertySymbols(object);
    enumerableOnly && (symbols = symbols.filter((function(sym) {
      return Object.getOwnPropertyDescriptor(object, sym).enumerable;
    }))), keys.push.apply(keys, symbols);
  }
  return keys;
}

function _objectSpread(target) {
  for (var i = 1; i < arguments.length; i++) {
    var source = null != arguments[i] ? arguments[i] : {};
    i % 2 ? ownKeys(Object(source), !0).forEach((function(key) {
      _defineProperty__default.default(target, key, source[key]);
    })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach((function(key) {
      Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
    }));
  }
  return target;
}

class SelectController extends Controller_dist_fields.default {
  constructor(_ref, ...args) {
    let {defaultValue: defaultValue = null} = _ref, config = _objectWithoutProperties__default.default(_ref, [ "defaultValue" ]);
    super(_objectSpread(_objectSpread({}, config), {}, {
      defaultValue: defaultValue
    }), ...args), _defineProperty__default.default(this, "getFilterGraphQL", ({value: {inverted: inverted, options: options}}) => {
      if (!options.length) return "";
      const isMulti = options.length > 1;
      let key = this.path;
      isMulti && inverted ? key = this.path + "_not_in" : isMulti ? key = this.path + "_in" : inverted && (key = this.path + "_not");
      return {
        [key]: isMulti ? options.map(x => x.value) : options[0].value
      };
    }), _defineProperty__default.default(this, "getFilterLabel", () => this.label), 
    _defineProperty__default.default(this, "formatFilter", ({value: value}) => {
      if (!value.options.length) return value.inverted ? this.label + " is set" : this.label + " has no value";
      if (value.options.length > 1) {
        const values = value.options.map(i => i.label).join(", ");
        return value.inverted ? `${this.label} is not in [${values}]` : `${this.label} is in [${values}]`;
      }
      const optionLabel = value.options[0].label;
      return value.inverted ? `${this.label} is not ${optionLabel}` : `${this.label} is ${optionLabel}`;
    }), _defineProperty__default.default(this, "getFilterValue", value => value && value.options && value.options.length ? value : void 0), 
    _defineProperty__default.default(this, "getFilterTypes", () => [ {
      type: "is",
      label: "Matches",
      getInitialValue: () => ({
        inverted: !1,
        options: []
      })
    } ]), this.options = config.options, this.dataType = config.dataType;
  }
}

exports.default = SelectController;
