// @keystonejs/fields/Controller/dist/fields.cjs.prod.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: !0
});

var _defineProperty = require("@babel/runtime/helpers/defineProperty"), _objectWithoutProperties = require("@babel/runtime/helpers/objectWithoutProperties"), _classCallCheck = require("@babel/runtime/helpers/classCallCheck"), isEqual = require("lodash.isequal");

function _interopDefault(e) {
  return e && e.__esModule ? e : {
    default: e
  };
}

var _defineProperty__default = _interopDefault(_defineProperty), _objectWithoutProperties__default = _interopDefault(_objectWithoutProperties), _classCallCheck__default = _interopDefault(_classCallCheck), isEqual__default = _interopDefault(isEqual), FieldController = function FieldController(_ref, _ref2, views) {
  var _this = this, readViews = _ref2.readViews, preloadViews = _ref2.preloadViews, getListByKey = _ref2.getListByKey, label = _ref.label, path = _ref.path, type = _ref.type, access = _ref.access, isOrderable = _ref.isOrderable, isPrimaryKey = _ref.isPrimaryKey, isRequired = _ref.isRequired, isReadOnly = _ref.isReadOnly, adminDoc = _ref.adminDoc, defaultValue = _ref.defaultValue, config = _objectWithoutProperties__default.default(_ref, [ "label", "path", "type", "access", "isOrderable", "isPrimaryKey", "isRequired", "isReadOnly", "adminDoc", "defaultValue" ]);
  _classCallCheck__default.default(this, FieldController), _defineProperty__default.default(this, "getQueryFragment", (function() {
    return _this.path;
  })), _defineProperty__default.default(this, "serialize", (function(data) {
    return data[_this.path] || null;
  })), _defineProperty__default.default(this, "validateInput", (function() {})), _defineProperty__default.default(this, "deserialize", (function(data) {
    return data[_this.path];
  })), _defineProperty__default.default(this, "hasChanged", (function(initialData, currentData) {
    return !isEqual__default.default(initialData[_this.path], currentData[_this.path]);
  })), _defineProperty__default.default(this, "getDefaultValue", (function() {
    var _ref3 = arguments.length > 0 && void 0 !== arguments[0] ? arguments[0] : {}, _ref3$originalInput = _ref3.originalInput, originalInput = void 0 === _ref3$originalInput ? {} : _ref3$originalInput, _ref3$prefill = _ref3.prefill, prefill = void 0 === _ref3$prefill ? {} : _ref3$prefill;
    return _this._getDefaultValue({
      originalInput: originalInput,
      prefill: prefill
    });
  })), _defineProperty__default.default(this, "initCellView", (function() {
    var Cell = _this.views.Cell;
    Cell && _this.readViews([ Cell ]);
  })), _defineProperty__default.default(this, "initFieldView", (function() {
    var Field = _this.views.Field;
    Field && _this.readViews([ Field ]);
  })), _defineProperty__default.default(this, "initFilterView", (function() {
    var Filter = _this.views.Filter;
    Filter && _this.readViews([ Filter ]);
  })), _defineProperty__default.default(this, "getFilterTypes", (function() {
    return [];
  })), _defineProperty__default.default(this, "getFilterValue", (function(value) {
    return value;
  })), this.config = config, this.label = label, this.path = path, this.type = type, 
  this.maybeAccess = access, this.isOrderable = isOrderable, this.isPrimaryKey = isPrimaryKey, 
  this.isRequired = isRequired, this.isReadOnly = isReadOnly, this.adminDoc = adminDoc, 
  this.readViews = readViews, this.preloadViews = preloadViews, this.getListByKey = getListByKey, 
  this.views = views, this._getDefaultValue = "function" != typeof defaultValue ? function(_ref4) {
    return _ref4.prefill[_this.path] || defaultValue;
  } : defaultValue;
};

exports.default = FieldController;
