// @keystonejs/fields/types/Relationship/views/Filter/dist/fields.esm.js
import '@babel/runtime/helpers/defineProperty';
import '@babel/runtime/helpers/objectWithoutProperties';
import '@emotion/core';
import React from 'react';
import '@babel/runtime/helpers/extends';
import '@apollo/client';
import '@arch-ui/select';
import 'react-select';
import 'intersection-observer';
import { R as RelationshipSelect } from '../../../../../dist/RelationshipSelect-9b9eafc5.esm.js';

const EventCatcher = ({
  children
}) => /*#__PURE__*/React.createElement("div", {
  onClick: e => {
    e.preventDefault();
    e.stopPropagation();
  }
}, children);

const RelationshipFilterView = ({
  onChange,
  filter,
  field,
  value
}) => {
  const handleChange = option => {
    if (option === null) {
      onChange(null);
    } else {
      const {
        value
      } = option;

      if (value) {
        onChange(value.id);
      }
    }
  };

  if (!filter) return null;
  const htmlID = `ks-input-${field.path}`;
  return /*#__PURE__*/React.createElement(EventCatcher, null, /*#__PURE__*/React.createElement(RelationshipSelect, {
    field: field,
    renderContext: null,
    htmlID: htmlID,
    onChange: handleChange,
    value: value,
    isMulti: false
  }));
};

export default RelationshipFilterView;
